programs/bpf_loader/src/serialization.rs

```diff
@@ -67,22 +72,38 @@ impl Serializer {
         unsafe {
             self.buffer.write_unchecked(value);
         }
+
+        vaddr
     }
 
-    fn write_all(&mut self, value: &[u8]) {
+    fn write_all(&mut self, value: &[u8]) -> u64 {
+        let vaddr = self
+            .vaddr
+            .saturating_add(self.buffer.len() as u64)
+            .saturating_sub(self.region_start as u64);
         // Safety:
         // see write() - the buffer is guaranteed to be large enough
         unsafe {
             self.buffer.write_all_unchecked(value);
         }
+
+        vaddr
     }
 
-    fn write_account(&mut self, account: &mut BorrowedAccount<'_>) -> Result<(), InstructionError> {
-        if self.copy_account_data {
+    fn write_account(
```

Comment:
I renamed push_account_region to push_account_data_region because I do think that's a bit more accurate. But write_account writes both the account data and the realloc region